app/scripts/modules/core/src/domain/IPipeline.ts

```diff
@@ -93,3 +94,8 @@ export interface IPipelineRef {
 export interface ITemplateInheritable {
   inherited?: boolean;
 }
+
+export interface IPipelineCategory {
+  name: string;
+  value: string;
+}
```

Comment:
Can you add some comment describing what this means? I would expect "PipelineCategory" to be a simple string or enum on a pipeline.  The name/value combination in this type is not immediately obvious.

Reply:
Is the confusion here perhaps because I am not simply allowing users to categorize their pipelines into categories defined as strings.

Rather, I am allowing pipeline authors to define category *names* and give their pipelines values for those categories?

For instance, if users wanted to group/filter by multiple dimensions, instead of them having to work around the system by defining categories:

- `"env=test"`
- `"env=prod"`
- `"region=east"`
- `"region=west"`
- `"user=alan"`
- `"user=chris"`

I tried to tackle this from the start by making multiple dimensions a first class concept here:

- So the category name "env" can have values "test" and "prod", the category name "region" can have values "east" and "west"
- And so on and so forth

Perhaps "category" is not the best term to describe this? See above screenshot again for clarification?

Reply:
Nonetheless, still good feedback. Is there a better way/term to make this more obvious?

Reply:
how about something more akin to FilterDimension or Metadata?  Category seems not quite right in this context.

Reply:
PipelineTags?

Reply:
Tags sound good.

Dimension seems to be the most technically accurate but I'm also thinking of how this is present to a user to configure in the pipeline config UI and Dimension doesn't seem obvious/intuitive enough.

Metadata is a little too broad or vague.